tests/nested/classic/hotplug/task.yaml

```diff
@@ -0,0 +1,297 @@
+
+    copy_remote "${GOHOME}"/snapd_*.deb
+    execute_remote "sudo apt update"
+    execute_remote "sudo apt install -y ./snapd_*.deb"
+    #shellcheck disable=SC2016
+    execute_remote 'DEBIAN_FRONTEND=noninteractive sudo -E apt install -y linux-image-extra-$(uname -r) || DEBIAN_FRONTEND=noninteractive sudo -E apt install -y linux-modules-extra-$(uname -r)'
+
+    snap pack "$TESTSLIB"/snaps/serial-port-hotplug
+    copy_remote serial-port-hotplug_1.0_all.snap
+    # add test user to dialout group required for ttyUSB* devices
+    execute_remote "sudo usermod -a -G dialout user1"
+
+    execute_remote "sudo snap install --devmode jq"
+
+restore: |
+    #shellcheck source=tests/lib/nested.sh
+    . "$TESTSLIB/nested.sh"
+    destroy_nested_vm
+    rm -rf /tmp/serialport{0,1}
+
+debug: |
+    #shellcheck source=tests/lib/nested.sh
+    . "$TESTSLIB/nested.sh"
+
+    execute_remote "snap connections --all" || true
+    execute_remote "snap list" || true
+    execute_remote "dmesg" || true
+    execute_remote 'sudo jq -r ".data[\"hotplug-slots\"] /var/lib/snapd/state.json' || true
+
+execute: |
+    hotplug_add_dev1() {
+        add_tty_chardev my-chardev /tmp/serialport1
+        add_usb_serial_device my-usb-serial my-chardev 1234
+    }
+
+    hotplug_del_dev1() {
+        del_device my-usb-serial
+        remove_chardev my-chardev
+    }
+
+    hotplug_add_dev2() {
+        add_tty_chardev my-chardev2 /tmp/serialport2
+        add_usb_serial_device my-usb-serial2 my-chardev2 5678
+    }
+
+    hotplug_del_dev2() {
+        del_device my-usb-serial2
+        remove_chardev my-chardev2
+    }
+
+    # Check that given slot is not present in 'snap connections' output
+    # (note, it can still be present in the state with hotplug-gone=true)
+    check_slot_not_present() {
+        SLOT_NAME="$1"
+        for _ in $(seq 10); do
+            if ! execute_remote "snap connections system" | MATCH ":$SLOT_NAME"; then
+                break
+            fi
+            sleep 1
+        done
+        if execute_remote "snap connections system" | MATCH ":$SLOT_NAME "; then
+            echo "slot $SLOT_NAME shouldn't be present anymore"
+            exit 1
+        fi
+    }
+
+    # Check that given slot is present in 'snap connections' output (but is not connected)
+    check_slot_present() {
+        SLOT_NAME="$1"
+        for _ in $(seq 10); do
+            if execute_remote "snap connections system" | MATCH "serial-port .* - .* :$SLOT_NAME"; then
+                break
+            fi
+            sleep 1
+        done
+        execute_remote "snap connections system" | MATCH "serial-port .* - .* :$SLOT_NAME"
+    }
+
+    # Check that given slot has hotplug-gone=true, meaning the device was unplugged but there are connections remembered for it
+    check_slot_gone() {
+        SLOT_NAME="$1"
+        execute_remote 'sudo jq -r ".data[\"hotplug-slots\"][\"'"$SLOT_NAME"'\"][\"hotplug-gone\"]" /var/lib/snapd/state.json' | MATCH "true"
```

Comment:
Yes, no disagreement, but we don't have that yet and it can become part of the story of `snap slots --devices ...` discussed briefly at the last sprint (potentially as a debug command as you suggested); I think it's a material for followup. I deliberately made it a helper shell function so should be easy to replace.